cEP-0009.md

```diff
@@ -0,0 +1,209 @@
+| Metadata     |                                                          |
+| ------------ |----------------------------------------------------------|
+| cEP          | 9                                                        |
+| Version      | 0.1                                                      |
+| Title        | Extract and utilize information from project files       |
+| Authors      | Satwik Kansal <satwikkansal@gmail.com>                   |
+| Status       | Proposed                                                 |
+| Type         | Feature                                                  |
+
+## Abstract
+
+There are many kinds of project files like ".editorconfig", "Gruntfile.js", etc. that contain meta-information that can used by tools like coala-quickstart to provide better analysis. This cEP proposes a mechnaism to represent and extract information from project files and an interface to utilize the extracted information.
+
+## Proposed Approach
+
+### 1. Implementation of the “Info” class
+
+A subclass of `Info` class will represent a specific kind of information. For example, the class `IgnorePathInfo` may represent “ignored file paths”.
+
+```py
+class Info:
+    description = "Some information"
+    # for validation
+    value_type = ("data-type of the information value, can also be an ``Enum``"
+                  "of possible choices.")
+
+    def __init__(self,
+                 source,
+                 value: value_type
+                 scope=None):
+        """
+        :param source: Source from where the information is discovered.
+        :param value:  Value of the infomation to be represented.
+        :param scope:  Scope of the applicability of the information. Some
```

Comment:
hmm, not sure. 

The basic idea was to define the areas where this information is valid and useful. I had another idea of adding bear names to "bear-specific" scope. For example
```py
scope: {
     "bear-specific" : ["CSSLintbear"]
}
```